src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -0,0 +1,403 @@
+            // RunLoop or StreamRef because IntPtrs are value types and can't be locked
+            lock (_lastWatchedDirectory)
+            {
+                // Always re-create the filter flags when start is called since they could have changed
+                if (_notifyFilters.HasFlag(NotifyFilters.Attributes | NotifyFilters.CreationTime | NotifyFilters.LastAccess | NotifyFilters.LastWrite | NotifyFilters.Size))
+                {
+                    _filterFlags = Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemInodeMetaMod | Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemFinderInfoMod;
+                }
+                if (_notifyFilters.HasFlag(NotifyFilters.Security))
+                {
+                    _filterFlags |= Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemChangeOwner | Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemXattrMod;
+                }
+                if (_notifyFilters.HasFlag(NotifyFilters.DirectoryName))
+                {
+                    _filterFlags |= Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemIsDir | 
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemIsSymlink | 
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemCreated | 
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemModified | 
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRemoved | 
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed;
+                }
+                if (_notifyFilters.HasFlag(NotifyFilters.FileName))
+                {
+                    _filterFlags |= Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemIsFile | 
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemIsSymlink |
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemCreated |
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemModified |
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRemoved |
+                                    Interop.CoreServices.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed;
+                }
+
+                // The directory has not changed then just restart receiving events. If the directory has changed 
+                // then we need to cleanup the old stream and kick off a new one
+                if (_lastWatchedDirectory.Equals(_directory, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
Makes sense.  Jon, for now, why don't you change the existing CaseSensitive constant to be a property, and change all of the call sites that do string comparisons to base it off of that property  We can figure out how to best address it separately, potentially by making that property return the appropriate sensitivity based on the watched directory's file system.